src/SymbolProvider/include/SymbolProvider/SymbolProvider.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SYMBOL_PROVIDER_SYMBOL_PROVIDER_H_
+#define SYMBOL_PROVIDER_SYMBOL_PROVIDER_H_
+
+#include "OrbitBase/CanceledOr.h"
+#include "OrbitBase/Result.h"
+#include "OrbitBase/StopToken.h"
+#include "SymbolProvider/ModuleIdentifier.h"
+
+namespace orbit_symbol_provider {
+
+// "SymbolProvider" supports retrieving symbols for the provided module from its symbol source, and
+// provides the local file path where symbols are cached if retrieval succeeded.
+// It also allows to query whether module symbols can be found in its symbol source.
+class SymbolProvider {
+ public:
+  virtual ~SymbolProvider() = default;
+
+  // Search for symbols for the provided module in the SymbolProvider's symbol source. Return true
+  // if found.
+  [[nodiscard]] virtual orbit_base::Future<bool> FindSymbols(const ModuleIdentifier& module_id) = 0;
```

Comment:
Looking at this again, I think the return type should be `orbit_base::Future<ErrorMessageOr<void>>`. Then in the not found case, it returns a message that explains why they were not found. This is then later used to display a detailed not found message to the user. (See `SymbolErrorDialog`). 

For your cases the message simply can be: "Not found in Stadia symbol store." or "Search in stadia symbol store not possible, because module does not contain a build id"